ultradns/zone.go

```diff
@@ -0,0 +1,255 @@
+	ResultInfo *ResultInfo     `json:"resultInfo"`
+	Zones      []*ZoneResponse `json:"zones"`
+}
+
+//create zone
+func (c *Client) CreateZone(zone *Zone) (*http.Response, error) {
+	target := Target(&SuccessResponse{})
+	res, err := c.Do(http.MethodPost, "zones", zone, target)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if res.StatusCode < http.StatusOK || res.StatusCode > http.StatusIMUsed {
+		return res, fmt.Errorf("error while creating a zone (%v) - %s", zone.Properties.Name, target.Error[0])
+	}
+
+	er := c.checkZoneTask(res)
+
+	if er != nil {
+		return res, er
+	}
+
+	return res, nil
+}
+
+//read zone
+func (c *Client) ReadZone(zoneName string) (*http.Response, *ZoneResponse, error) {
+	target := Target(&ZoneResponse{})
+	zoneName = strings.Replace(zoneName, "/", "%2F", 1)
+	res, err := c.Do(http.MethodGet, "zones/"+zoneName, nil, target)
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if res.StatusCode < http.StatusOK || res.StatusCode > http.StatusIMUsed {
+		return res, nil, fmt.Errorf("error while reading a zone (%v) - %s", zoneName, target.Error[0])
+	}
+	zoneResponse := target.Data.(*ZoneResponse)
+
+	return res, zoneResponse, nil
+}
+
+//update zone
+func (c *Client) UpdateZone(zoneName string, zone *Zone) (*http.Response, error) {
+	target := Target(&SuccessResponse{})
+	zoneName = strings.Replace(zoneName, "/", "%2F", 1)
+	res, err := c.Do(http.MethodPut, "zones/"+zoneName, zone, target)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if res.StatusCode < http.StatusOK || res.StatusCode > http.StatusIMUsed {
+		return res, fmt.Errorf("error while updating a zone (%v) - %s", zoneName, target.Error[0])
+	}
+
+	return res, nil
+}
+
+//partial update zone
+func (c *Client) PatchUpdateZone(zoneName string, jsonPatch []*JsonPatch) (*http.Response, error) {
+	target := Target(&SuccessResponse{})
+	zoneName = strings.Replace(zoneName, "/", "%2F", 1)
+	res, err := c.Do(http.MethodPatch, "zones/"+zoneName, jsonPatch, target)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if res.StatusCode < http.StatusOK || res.StatusCode > http.StatusIMUsed {
+		return res, fmt.Errorf("error while updating a zone (%v) - %s", zoneName, target.Error[0])
+	}
+
+	return res, nil
+}
+
+//delete zone
+func (c *Client) DeleteZone(zoneName string) (*http.Response, error) {
+	target := Target(&SuccessResponse{})
+	zoneName = strings.Replace(zoneName, "/", "%2F", 1)
+	res, err := c.Do(http.MethodDelete, "zones/"+zoneName, nil, target)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if res.StatusCode < http.StatusOK || res.StatusCode > http.StatusIMUsed {
+		return res, fmt.Errorf("error while Deleting a zone (%v) - %s", zoneName, target.Error[0])
+	}
+
+	return res, nil
+}
+
+//list Zones
+func (c *Client) ListZone(query string) (*http.Response, *ZoneListResponse, error) {
+	target := Target(&ZoneListResponse{})
+	query = strings.Replace(query, "/", "%2F", 1)
+	res, err := c.Do(http.MethodGet, "zones/"+query, nil, target)
```

Comment:
I think we should be calling /v3 version of zones as it would list zones from all the internal platforms